src/borg/testsuite/crypto.py

```diff
@@ -228,3 +233,57 @@ def test_hkdf_hmac_sha512_5(self):
 
         okm = hkdf_hmac_sha512(ikm, salt, info, l)
         assert okm == bytes.fromhex('1407d46013d98bc6decefcfee55f0f90b0c7f63d68eb1a80eaf07e953cfc0a3a5240a155d6e4daa965bb')
+
+
+def test_decrypt_key_file_v2(monkeypatch):
+    # Arrange
+    plain = b'hello'
+    enc_key = bytes.fromhex('b668a71dded9a9772686ab4c2da2a06c491087e9e27c14f9e798ad8b929506b4')
+    cypher = AES(enc_key, b'\0'*16).encrypt(plain)
+    mac_key = bytes.fromhex('3ef3878e4bb03055b6efbecddcdbfc58bfc1617a0e1790a46c1a9991c3269b34')
+    hash = hmac_sha256(mac_key, cypher)
+    encrypted = msgpack.packb({
+        'version': 2,
+        'salt': b'salt'*16,
+        'argon2_time_cost': 1,
+        'argon2_memory_cost': 4096,
+        'argon2_parallelism': 1,
+        'argon2_type': b'I',
+        'algorithm': 'sha256',
+        'hash': hash,
+        'data': cypher
+    })
+    monkeypatch.setenv('BORG_PASSPHRASE', "hello, pass phrase")
+    passphrase = Passphrase.new()
+    key = KeyfileKey(None)
+
+    # Act
+    decrypted = key.decrypt_key_file(encrypted, passphrase)
+
+    # Assert
+    assert decrypted == plain
+
+
+def test_decrypt_key_file_v2_invalid_mac(monkeypatch):
+    # Arrange
+    monkeypatch.setenv('BORG_PASSPHRASE', "hello, pass phrase")
+    passphrase = Passphrase.new()
+    key = KeyfileKey(None)
+    data = b'hmac me'
+    encrypted = msgpack.packb({
+        'version': 2,
+        'salt': b'salt'*16,
+        'argon2_time_cost': 1,
+        'argon2_memory_cost': 4096,
+        'argon2_parallelism': 1,
+        'argon2_type': b'I',
+        'algorithm': 'sha256',
+        'hash': b'E'*32,
+        'data': data
+    })
+
+    # Act
+    result = key.decrypt_key_file(encrypted, passphrase)
+
+    # Assert
+    assert result is None
```

Comment:
hmm, do you think arrange / act / assert comments are giving enough value to have them?
it's always that order and also the assert is rather obvious anyway.